pkg/cloudprovider/providers/vsphere/nodemanager.go

```diff
@@ -257,6 +260,38 @@ func (nm *NodeManager) removeNode(node *v1.Node) {
 	nm.nodeInfoLock.Unlock()
 }
 
+func (nm *NodeManager) RegisterAttachedDisk(nodeName k8stypes.NodeName, volPath string) error {
+	nm.attachedDisksLock.Lock()
+	if _, ok := nm.attachedDisks[convertToString(nodeName)]; !ok {
+		nm.attachedDisks[convertToString(nodeName)] = make(map[string]bool)
+	}
+	nm.attachedDisks[convertToString(nodeName)][volPath] = true
+	nm.attachedDisksLock.Unlock()
+	return nil
+}
+func (nm *NodeManager) UnRegisterAttachedDisk(nodeName k8stypes.NodeName, volPath string) error {
+	nm.attachedDisksLock.Lock()
+	delete(nm.attachedDisks[convertToString(nodeName)], volPath)
+	if len(nm.attachedDisks[convertToString(nodeName)]) == 0 {
+		delete(nm.attachedDisks, convertToString(nodeName))
+	}
+	nm.attachedDisksLock.Unlock()
+	return nil
+}
+func (nm *NodeManager) GetAttachedDisks(nodeName k8stypes.NodeName) []string {
+	nm.attachedDisksLock.Lock()
+	attachedDisks := nm.attachedDisks[convertToString(nodeName)]
+	nm.attachedDisksLock.Unlock()
```

Comment:
I think attachedDisks needs to be protected with locks. Since this is the same object which will be accessed by  RegisterAttachedDisk and UnRegisterAttachedDisk